src/System.Text.Json/tests/Serialization/Value.ReadTests.NonGenericCollections.cs

```diff
@@ -305,7 +305,7 @@ public static void ReadPrimitiveStack()
             Assert.Equal(0, count);
 
             // TODO: use reflection to support types deriving from Stack.
-            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<StackWrapper>(@"[1,2]"));
+            Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize<StackWrapper>(@"[1,2]"));
```

Comment:
Stack isn't a read-only collection. How come we ended up changing the exception type for these tests?

Reply:
Per offline conversation:

There are other reasons apart from being readonly why an implementing type may not be supported.
 
For example,

```csharp
public StackWrapper : Stack {}
```

We currently don't support it because we don't have a .csproj ref to System.Collections.NonGeneric, so we can't cast StackWrapper to Stack and call "Push" to populate. To support it, we'd have to use reflection to probe for the right constructor.

`StackWrapper` and other unsupported types should now throw `NotSupportedException`.